veloroutes_voies_vertes/processing/config/save_style_database.py

```diff
@@ -0,0 +1,78 @@
+__author__ = '3liz'
+__date__ = '2019-02-15'
+__copyright__ = '(C) 2019 by 3liz'
+
+# This will get replaced with a git SHA1 when you do a git archive
+
+__revision__ = '$Format:%H$'
+
+from qgis.core import (
+    QgsProcessingParameterBoolean,
+    QgsProcessingOutputString,
+    QgsProject,
+)
+
+from ...qgis_plugin_tools.tools.algorithm_processing import BaseProcessingAlgorithm
+
+
+class SaveStyleDatabase(BaseProcessingAlgorithm):
+
+    CONFIRM_SAVE = 'CONFIRM_SAVE'
+    OUTPUT_STRING = 'OUTPUT_STRING'
+
+    def name(self):
+        return 'save_style_database'
+
+    def displayName(self):
+        return 'Sauvegerder les styles en base de données'
+
+    def group(self):
+        return 'Configuration'
+
+    def groupId(self):
+        return 'veloroutes_configuration'
+
+    def shortHelpString(self) -> str:
+        return 'Sauvegarde les styles en base de données'
+
+    def initAlgorithm(self, config):
+        """
+        Here we define the inputs and output of the algorithm, along
+        with some other properties.
+        """
+        # INPUTS
+
+        # Boolean to confirm save
+        input_bool = QgsProcessingParameterBoolean(self.CONFIRM_SAVE, 'Confirmez la sauvegarde des styles')
+        self.addParameter(input_bool)
+
+        # OUTPUTS
+
+        # Add output for message
+        output = QgsProcessingOutputString(self.OUTPUT_STRING, 'Message de sortie')
+        output.tooltip_3liz = output.description()
+        self.addOutput(output)
+
+    def processAlgorithm(self, parameters, context, feedback):
+        """
+        Here is where the processing itself takes place.
+        """
+        comfirm = self.parameterAsBool(parameters, self.CONFIRM_SAVE, context)
+
+        if not comfirm:
+            return {
+                self.OUTPUT_STRING: 'les styles n\'ont pas été sauvegardés'
+            }
+
+        layers = QgsProject.instance().mapLayers()
+        if len(layers) < 1:
+            return {
+                self.OUTPUT_STRING: 'Aucune couche n\'a été trouvés'
+            }
+        for lay in layers:
+            layer = QgsProject.instance().mapLayer(lay)
```

Comment:
```suggestion
        for lay in layers.values():
```
Il y a déjà un appel au projet juste au-dessus qui te redonne déjà la liste des couches format QgsMapLayer